drivers/dac/dac_ad569x.c

```diff
@@ -0,0 +1,239 @@
+	AD569X_CTRL_PD_1K_OUT_IMPEDANCE = 1,
+	AD569X_CTRL_PD_100K_OUT_IMPEDANCE = 2,
+	AD569X_CTRL_PD_3_STATE_OUT_IMPEDANCE = 3,
+	AD569X_CTRL_PD_DEFAULT = AD569X_CTRL_PD_NORMAL,
+	AD569X_CTRL_PD_MAX = AD569X_CTRL_PD_3_STATE_OUT_IMPEDANCE,
+};
+
+enum {
+	AD569X_CTRL_NO_RESET = 0,
+	AD569X_CTRL_PERFORM_RESET = 1,
+	AD569X_CTRL_RESET_DEFAULT = AD569X_CTRL_NO_RESET
+};
+
+static int ad569x_write(const struct device *dev, enum ad569x_command command, uint16_t value)
+{
+	const struct ad569x_config *config = dev->config;
+
+	uint8_t tx_data[3];
+
+	tx_data[0] = command;
+	sys_put_be16(value, tx_data + 1);
+
+	return i2c_write_dt(&config->bus, tx_data, sizeof(tx_data));
+}
+
+static int ad569x_read(const struct device *dev, uint16_t *value)
+{
+	const struct ad569x_config *config = dev->config;
+
+	uint8_t rx_data[2];
+	int result;
+
+	result = i2c_read_dt(&config->bus, rx_data, sizeof(rx_data));
+	if (result == 0) {
+		*value = sys_get_be16(rx_data);
+	}
+
+	return result;
+}
+
+static int ad569x_channel_setup(const struct device *dev, const struct dac_channel_cfg *channel_cfg)
+{
+	const struct ad569x_config *config = dev->config;
+
+	if (channel_cfg->channel_id > DAC_AD569X_MAX_CHANNEL) {
+		LOG_ERR("invalid channel %d", channel_cfg->channel_id);
+		return -EINVAL;
+	}
+
+	if (channel_cfg->resolution != config->resolution) {
+		LOG_ERR("invalid resolution %d", channel_cfg->resolution);
+		return -EINVAL;
+	}
+
+	if (config->gain > AD569X_CTRL_GAIN_MAX) {
+		LOG_ERR("invalid gain %d", config->gain);
+		return -EINVAL;
+	}
+
+	if (config->voltage_reference > AD569X_CTRL_REF_MAX) {
+		LOG_ERR("invalid voltage reference %d", config->voltage_reference);
+		return -EINVAL;
+	}
+
+	if (config->power_down_mode > AD569X_CTRL_PD_MAX) {
+		LOG_ERR("invalid power down mode %d", config->power_down_mode);
+		return -EINVAL;
+	}
+
+	LOG_DBG("configure %s: gain %d, voltage reference %d, power down mode %d", dev->name,
+		config->gain, config->voltage_reference, config->power_down_mode);
+
+	uint16_t ctrl_reg = AD569X_CTRL_GAIN(config->gain) |
+			    AD569X_CTRL_REF(config->voltage_reference) |
+			    AD569X_CTRL_PD(config->power_down_mode);
+
+	int result = ad569x_write(dev, AD569X_CMD_CONFIGURE, ctrl_reg);
+
+	if (result != 0) {
+		LOG_ERR("failed to configure the device");
+		return result;
+	}
+
+	return 0;
+}
+
+static int ad569x_sw_reset(const struct device *dev)
+{
+	uint16_t reg = AD569X_CTRL_RESET(AD569X_CTRL_PERFORM_RESET);
+
+	LOG_DBG("reset %s", dev->name);
+
+	/* Ignore return value, since device gives NAK after receiving RESET request */
+	ad569x_write(dev, AD569X_CMD_CONFIGURE, reg);
+
+	/* Check that DAC output is reset */
+	int result = ad569x_read(dev, &reg);
+
+	if (result != 0) {
+		LOG_ERR("failed to read value");
+		return result;
+	}
+
+	if (reg != 0) {
+		LOG_ERR("failed to reset DAC output");
+		result = -EIO;
+	}
+
+	return 0;
+}
+
+static int ad569x_set_output(const struct device *dev, uint8_t channel, uint32_t value)
```

Comment:
For consistency with the public API:
```suggestion
static int ad569x_write_value(const struct device *dev, uint8_t channel, uint32_t value)
```